packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -986,6 +989,23 @@ class TextSelectionGestureDetectorBuilder {
     _shouldShowSelectionToolbar = kind == null
       || kind == PointerDeviceKind.touch
       || kind == PointerDeviceKind.stylus;
+
+    // Handle shift + click selection if needed.
+    if (editableText.isShiftPressed && renderEditable.selection?.baseOffset != null) {
+      _isShiftTapping = true;
+      switch (defaultTargetPlatform) {
+        case TargetPlatform.iOS:
+        case TargetPlatform.macOS:
+          renderEditable.expandSelection(cause: SelectionChangedCause.tap);
```

Comment:
Does the current `RenderEditable` interface expose a method that converts global coordinates to `TextPosition`? If there is then maybe we can implement `expandSelection` here as a private method? Eventually I think we'd like to remove all the mutating methods on `RenderEditable` right? So adding new mutable APIs to it increases the chance that the API removal becomes a breaking change.

Reply:
Looks like yes, `getPositionForPoint`, and then I can set the selection with `editableText.userUpdateTextEditingValue`.  Thanks for the suggestion, this was the biggest thing bugging me about my approach.